src/config.rs

```diff
@@ -33,44 +34,123 @@ impl Shell {
     }
 }
 
-#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, PartialEq)]
 pub(crate) struct FlokiConfig {
```

Comment:
I'm not completely sold on having two different config structs - it seems like a lot of duplication, especially when new `floki.yaml` options are added. However, the only alternatives I could think of we more unpleasant:
- Implementing `Deserialize` for `FlokiConfig` rather than deriving it for `StaticFlokiConfig`, and having it read some options from the file and compute others;
- Having `FlokiConfig` store a `FlokiStaticConfig` struct, thus forcing clients to know which options are on the `FlokiConfig` struct and which are on the `FlokiStaticConfig` one.

This middle-ground at least gets us automatic implementation of `Deserialize` and a flat `FlokiConfig` to expose to the rest of the application. Suggestions for something better are welcome.

Reply:
Yes, I'm not sure about this - I think essentially your question boils down to where things live in general.

Here is my steer, tell me what you think!

`FlokiConfig` ought to contain the contents of a parsed (logical) `floki.yaml`.

Everything else that depends on how `floki` is launched ought to live in the `Environment` structure. So I think that means `Environment` will need fields for a) the directory on the host that is being mounted (where `floki.yaml` lives), and b) the directory we are currently in. We ought to be able to derive the rest in some functions of `FlokiConfig` and `Environment`.

I admit, the naming of these structs might not be optimal.

So, one way to put it is, `FlokiConfig` captures the contents of a config file, `Environment` captures how `floki` was launched, `Cli` captures the CLI, and the rest we do through munging and functions.

Reply:
So I think this is the main point I'm not sure about too, so I'll hold off going into too much detail for when we've figured out what we want here. What do you think to might idea above?

Let's figure this bit out, and see how everything else hangs together.

Reply:
Just to check I'm not sending you into a firey abyss, I at least had the sense to make `Environment` not depend on `FlokiConfig`. This means we can build it before reading the config file.

Reply:
What you've suggested feels like a much better split to me. I'll make this change and see how it looks in the code 😺

Reply:
I've now made this change, and I think it's a big improvement - it seems to "click" much better than the two config structs, and this is reflected in a smaller diff. Let me know what you think.

Reply:
Yes, this looks pretty natural now! I'll take a proper look over the weekend, but I suspect further comments from me will be more nitpicky. I want to convince myself everything works as I expect, and then I'll be happy.